// @stacksjs/path/dist/index.cjs
'use strict';

const pathe = require('pathe');

function aiPath(path2) {
  return corePath(`ai/${path2 || ""}`);
}
function actionsPath(path2) {
  return corePath(`actions/src/${path2 || ""}`);
}
function aliasPath() {
  return corePath("alias/src/index.ts");
}
function runtimePath(path2) {
  return frameworkPath(`buddy/${path2 || ""}`);
}
function arraysPath(path2) {
  return corePath(`arrays/${path2 || ""}`);
}
function authPath(path2) {
  return corePath(`auth/${path2 || ""}`);
}
function appPath(path2) {
  return projectPath(`app/${path2 || ""}`);
}
function buildPath(path2) {
  return corePath(`build/${path2 || ""}`);
}
function buildEnginePath(path2) {
  return buildPath(`${path2 || ""}`);
}
function buildEntriesPath(path2) {
  return buildPath(`entries/${path2 || ""}`);
}
function cachePath(path2) {
  return corePath(`cache/${path2 || ""}`);
}
function chatPath(path2) {
  return corePath(`chat/${path2 || ""}`);
}
function cliPath(path2) {
  return corePath(`cli/${path2 || ""}`);
}
function cloudPath(path2) {
  return corePath(`cloud/${path2 || ""}`);
}
function collectionsPath(path2) {
  return corePath(`collections/${path2 || ""}`);
}
function componentsPath(path2) {
  return projectPath(`components/${path2 || ""}`);
}
function configPath(path2) {
  return corePath(`config/${path2 || ""}`);
}
function userConfigPath(path2) {
  return projectPath(`config/${path2 || ""}`);
}
function corePath(path2) {
  return frameworkPath(`core/${path2 || ""}`);
}
function dashboardPath(path2) {
  return corePath(`dashboard/${path2 || ""}`);
}
function databasePath(path2) {
  return corePath(`database/${path2 || ""}`);
}
function developmentPath(path2) {
  return corePath(`development/${path2 || ""}`);
}
function desktopPath(path2) {
  return corePath(`desktop/${path2 || ""}`);
}
function docsPath(path2) {
  return corePath(`docs/${path2 || ""}`);
}
function domainsPath(path2) {
  return corePath(`domains/${path2 || ""}`);
}
function customElementsDataPath() {
  return frameworkPath("custom-elements.json");
}
function emailPath(path2) {
  return corePath(`email/${path2 || ""}`);
}
function errorHandlingPath(path2) {
  return corePath(`error-handling/${path2 || ""}`);
}
function eventsPath(path2) {
  return corePath(`events/${path2 || ""}`);
}
function fakerPath(path2) {
  return corePath(`faker/${path2 || ""}`);
}
function healthPath(path2) {
  return corePath(`health/${path2 || ""}`);
}
function examplesPath(type) {
  return frameworkPath(`examples/${type || ""}`);
}
function frameworkPath(path2) {
  return projectPath(`.stacks/${path2 || ""}`);
}
function storagePath(path2) {
  return corePath(`storage/${path2 || ""}`);
}
function functionsPath(path2) {
  return projectPath(`functions/${path2 || ""}`);
}
function gitPath(path2) {
  return corePath(`git/${path2 || ""}`);
}
function langPath(path2) {
  return projectPath(`lang/${path2 || ""}`);
}
function libraryEntryPath(type) {
  return buildEntriesPath(`${type}.ts`);
}
function lintPath(path2) {
  return corePath(`lint/${path2 || ""}`);
}
function loggingPath(path2) {
  return corePath(`logging/${path2 || ""}`);
}
function xRayPath(path2) {
  return corePath(`x-ray/${path2 || ""}`);
}
function modelsPath(path2) {
  return appPath(`models/${path2 || ""}`);
}
function modulesPath(path2) {
  return corePath(`modules/${path2 || ""}`);
}
function notificationsPath(path2) {
  return corePath(`notifications/${path2 || ""}`);
}
function ormPath(path2) {
  return corePath(`orm/${path2 || ""}`);
}
function objectsPath(path2) {
  return corePath(`objects/${path2 || ""}`);
}
function onboardingPath(path2) {
  return projectPath(`${path2 || "pages/dashboard/onboarding"}`);
}
function packageJsonPath(type) {
  if (type === "vue-components")
    return frameworkPath("components/vue/package.json");
  if (type === "web-components")
    return frameworkPath("components/web/package.json");
  if (type === "functions")
    return frameworkPath("functions/package.json");
  return frameworkPath(`${type}/package.json`);
}
function pagesPath(path2) {
  return appPath(`pages/${path2 || ""}`);
}
function pathPath(path2) {
  return corePath(`path/${path2 || ""}`);
}
function paymentsPath(path2) {
  return corePath(`payments/${path2 || ""}`);
}
function projectPath(filePath = "") {
  let path2 = process.cwd();
  while (path2.includes(".stacks"))
    path2 = pathe.resolve(path2, "..");
  return pathe.resolve(path2, filePath);
}
function pushPath(path2) {
  return corePath(`push/${path2 || ""}`);
}
function queryBuilderPath(path2) {
  return corePath(`query-builder/${path2 || ""}`);
}
function queuePath(path2) {
  return corePath(`queue/${path2 || ""}`);
}
function realtimePath(path2) {
  return corePath(`realtime/${path2 || ""}`);
}
function routerPath(path2) {
  return corePath(`router/${path2 || ""}`);
}
function searchEnginePath(path2) {
  return corePath(`search-engine/${path2 || ""}`);
}
function settingsPath(path2) {
  return projectPath(`${path2 || "pages/dashboard/settings"}`);
}
function smsPath(path2) {
  return corePath(`sms/${path2 || ""}`);
}
function schedulerPath(path2) {
  return corePath(`scheduler/${path2 || ""}`);
}
function signalsPath(path2) {
  return corePath(`signals/${path2 || ""}`);
}
function routesPath(path2) {
  return projectPath(`routes/${path2 || ""}`);
}
function securityPath(path2) {
  return corePath(`security/${path2 || ""}`);
}
function serverPath(path2) {
  return corePath(`server/${path2 || ""}`);
}
function serverlessPath(path2) {
  return corePath(`serverless/${path2 || ""}`);
}
function stacksPath(path2) {
  return frameworkPath(`src/${path2 || ""}`);
}
function tablesPath(path2) {
  return corePath(`tables/${path2 || ""}`);
}
function testingPath(path2) {
  return corePath(`testing/${path2 || ""}`);
}
function testsPath(path2) {
  return frameworkPath(`tests/${path2 || ""}`);
}
function typesPath(path2) {
  return corePath(`types/${path2 || ""}`);
}
function stringsPath(path2) {
  return corePath(`strings/${path2 || ""}`);
}
function scriptsPath(path2) {
  return frameworkPath(`scripts/${path2 || ""}`);
}
function uiPath(path2) {
  return corePath(`ui/${path2 || ""}`);
}
function utilsPath(path2) {
  return corePath(`utils/${path2 || ""}`);
}
function validationPath(path2) {
  return corePath(`validation/${path2 || ""}`);
}
const path = {
  aiPath,
  actionsPath,
  aliasPath,
  arraysPath,
  appPath,
  authPath,
  buildEnginePath,
  buildEntriesPath,
  buildPath,
  cachePath,
  chatPath,
  cliPath,
  cloudPath,
  collectionsPath,
  componentsPath,
  configPath,
  userConfigPath,
  corePath,
  customElementsDataPath,
  databasePath,
  developmentPath,
  dashboardPath,
  desktopPath,
  docsPath,
  domainsPath,
  emailPath,
  errorHandlingPath,
  eventsPath,
  healthPath,
  examplesPath,
  fakerPath,
  frameworkPath,
  storagePath,
  functionsPath,
  gitPath,
  langPath,
  libraryEntryPath,
  lintPath,
  loggingPath,
  modulesPath,
  ormPath,
  objectsPath,
  onboardingPath,
  notificationsPath,
  packageJsonPath,
  pagesPath,
  pathPath,
  paymentsPath,
  projectPath,
  pushPath,
  queryBuilderPath,
  queuePath,
  realtimePath,
  routerPath,
  routesPath,
  searchEnginePath,
  schedulerPath,
  settingsPath,
  smsPath,
  signalsPath,
  scriptsPath,
  securityPath,
  serverPath,
  serverlessPath,
  stacksPath,
  stringsPath,
  tablesPath,
  testingPath,
  testsPath,
  typesPath,
  uiPath,
  utilsPath,
  validationPath,
  xRayPath,
  // path utils
  basename: pathe.basename,
  delimiter: pathe.delimiter,
  dirname: pathe.dirname,
  extname: pathe.extname,
  format: pathe.format,
  isAbsolute: pathe.isAbsolute,
  join: pathe.join,
  normalize: pathe.normalize,
  normalizeString: pathe.normalizeString,
  parse: pathe.parse,
  relative: pathe.relative,
  resolve: pathe.resolve,
  sep: pathe.sep,
  toNamespacedPath: pathe.toNamespacedPath
};

exports.basename = pathe.basename;
exports.delimiter = pathe.delimiter;
exports.dirname = pathe.dirname;
exports.extname = pathe.extname;
exports.format = pathe.format;
exports.isAbsolute = pathe.isAbsolute;
exports.join = pathe.join;
exports.normalize = pathe.normalize;
exports.normalizeString = pathe.normalizeString;
exports.parse = pathe.parse;
exports.relative = pathe.relative;
exports.resolve = pathe.resolve;
exports.sep = pathe.sep;
exports.toNamespacedPath = pathe.toNamespacedPath;
exports.actionsPath = actionsPath;
exports.aiPath = aiPath;
exports.aliasPath = aliasPath;
exports.appPath = appPath;
exports.arraysPath = arraysPath;
exports.authPath = authPath;
exports.buildEnginePath = buildEnginePath;
exports.buildEntriesPath = buildEntriesPath;
exports.buildPath = buildPath;
exports.cachePath = cachePath;
exports.chatPath = chatPath;
exports.cliPath = cliPath;
exports.cloudPath = cloudPath;
exports.collectionsPath = collectionsPath;
exports.componentsPath = componentsPath;
exports.configPath = configPath;
exports.corePath = corePath;
exports.customElementsDataPath = customElementsDataPath;
exports.dashboardPath = dashboardPath;
exports.databasePath = databasePath;
exports.desktopPath = desktopPath;
exports.developmentPath = developmentPath;
exports.docsPath = docsPath;
exports.domainsPath = domainsPath;
exports.emailPath = emailPath;
exports.errorHandlingPath = errorHandlingPath;
exports.eventsPath = eventsPath;
exports.examplesPath = examplesPath;
exports.fakerPath = fakerPath;
exports.frameworkPath = frameworkPath;
exports.functionsPath = functionsPath;
exports.gitPath = gitPath;
exports.healthPath = healthPath;
exports.langPath = langPath;
exports.libraryEntryPath = libraryEntryPath;
exports.lintPath = lintPath;
exports.loggingPath = loggingPath;
exports.modelsPath = modelsPath;
exports.modulesPath = modulesPath;
exports.notificationsPath = notificationsPath;
exports.objectsPath = objectsPath;
exports.onboardingPath = onboardingPath;
exports.ormPath = ormPath;
exports.packageJsonPath = packageJsonPath;
exports.pagesPath = pagesPath;
exports.path = path;
exports.pathPath = pathPath;
exports.paymentsPath = paymentsPath;
exports.projectPath = projectPath;
exports.pushPath = pushPath;
exports.queryBuilderPath = queryBuilderPath;
exports.queuePath = queuePath;
exports.realtimePath = realtimePath;
exports.routerPath = routerPath;
exports.routesPath = routesPath;
exports.runtimePath = runtimePath;
exports.schedulerPath = schedulerPath;
exports.scriptsPath = scriptsPath;
exports.searchEnginePath = searchEnginePath;
exports.securityPath = securityPath;
exports.serverPath = serverPath;
exports.serverlessPath = serverlessPath;
exports.settingsPath = settingsPath;
exports.signalsPath = signalsPath;
exports.smsPath = smsPath;
exports.stacksPath = stacksPath;
exports.storagePath = storagePath;
exports.stringsPath = stringsPath;
exports.tablesPath = tablesPath;
exports.testingPath = testingPath;
exports.testsPath = testsPath;
exports.typesPath = typesPath;
exports.uiPath = uiPath;
exports.userConfigPath = userConfigPath;
exports.utilsPath = utilsPath;
exports.validationPath = validationPath;
exports.xRayPath = xRayPath;
